Km.demo 403 track file utility add state

kiley-mitti opened the pull request:
[DEMO-403](https://rocketcom.atlassian.net/browse/DEMO-403)

## Brief Description

This PR adds state to Track Data Utility track list.
Additionally I needed to change the way mock data generates the creationDate on the track files so that they were not all the same day.
Additionally I tweaked the way that track files get named so they do not all have the same name.

I also switched the way that a single track file is selected, and simplified some of the functions.
I unhooked the checkboxes from track file selection.

Incorporated table service

## Issues and/or Limitations

- Some confusion about what checkboxes are supposed to actually do, need to confer with design. for now they do nothing
- Ditto editing tracks.. for now I've added a fake 'comment' field that is attached to TrackFile in App State

## Final Checklist

- [x] Works in Chrome
- [x] Works in Firefox
- [x] Works in Safari
- [x] Handles responsiveness as required
- [x] Dark theme styles are correct
- [x] Light theme styles are correct

brikiernan reviewed (commented):
@kiley-mitti there is a [table-service](https://github.com/RocketCommunicationsInc/Flight-Dynamics-Service/blob/main/src/app/shared/table.service.ts) which can handle all the sorting and selecting for this component. There are a couple of examples of it being used in the app. I'd be happy to connect with you in the morning to go over it a bit if you'd like.

NicDominguez reviewed (changes requested):
I believe its worth implementing the table service for this to clean up streamline the app. I just merged an example in the propagator utility and Brian has one in the solution table. Please connect with Brian if you need assistance implementing it. Its a little complex to work with but it will clean up the component a bunch.

kiley-mitti commented:
Added the table service with a little extension of my own to handle filtering the track files. Also did a lot of selector cleanup and paring down subscriptions. Added the destroy method of subscription cleanup that Jacob recommended
